searx/plugins/ahmia_filter.py

```diff
@@ -1,34 +1,51 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 # pylint: disable=missing-module-docstring
-
 from __future__ import annotations
+import typing
 from hashlib import md5
 
-import flask
+from flask_babel import gettext
 
 from searx.data import ahmia_blacklist_loader
 from searx import get_setting
+from searx.plugins import Plugin, PluginInfo
 
-
-name = "Ahmia blacklist"
-description = "Filter out onion results that appear in Ahmia's blacklist. (See https://ahmia.fi/blacklist)"
-default_on = True
-preference_section = 'onions'
+if typing.TYPE_CHECKING:
+    import flask
+    from searx.search import SearchWithPlugins
+    from searx.extended_types import SXNG_Request
+    from searx.result_types import Result
+    from searx.plugins import PluginCfg
 
 ahmia_blacklist: list = []
 
 
-def on_result(_request, _search, result) -> bool:
-    if not getattr(result, 'is_onion', None) or not getattr(result, 'parsed_url', None):
+class SXNGPlugin(Plugin):
+    """Filter out onion results that appear in Ahmia's blacklist (See https://ahmia.fi/blacklist)."""
+
+    id = "ahmia_filter"
+
+    def __init__(self, plg_cfg: "PluginCfg") -> None:
+        super().__init__(plg_cfg)
+        self.info = PluginInfo(
+            id=self.id,
+            name=gettext("Ahmia blacklist"),
+            description=gettext("Filter out onion results that appear in Ahmia's blacklist."),
+            preference_section="general",
+        )
+
+    def on_result(
+        self, request: "SXNG_Request", search: "SearchWithPlugins", result: Result
+    ) -> bool:  # pylint: disable=unused-argument
+        if not getattr(result, "is_onion", None) or not getattr(result, "parsed_url", None):
```

Comment:
```suggestion
        if not getattr(result, "is_onion", False) or not getattr(result, "parsed_url", None):
```
This might help to understand it better that the default is not onion.